Upstream: Shouldn't be, maybe musl or a compat lib should implement it
Reason: Fixes compilation on musl

diff --git a/pipewire-pulseaudio/src/stream.c b/pipewire-pulseaudio/src/stream.c
index 18fff46..1d8311a 100644
--- a/pipewire-pulseaudio/src/stream.c
+++ b/pipewire-pulseaudio/src/stream.c
@@ -40,6 +40,18 @@
 #define MAX_BUFFER_SAMPLES	(8*1024u)
 #define MAX_SIZE		(4*1024*1024u)
 
+/* strndupa may already be defined in another compatibility header */
+#if !defined(strndupa)
+#define strndupa(x_src, x_n) \
+        (__extension__ ( {   \
+                const char* x_in  = (x_src);                  \
+                size_t      x_len = strnlen(x_in, (x_n)) + 1; \
+                char*       x_out = (char *) alloca(x_len);   \
+                x_out[x_len-1] = '\0';                        \
+                (char *) memcpy(x_out, x_in, x_len-1);        \
+        } ) )
+#endif
+
 static void dump_buffer_attr(pa_stream *s, pa_buffer_attr *attr)
 {
 	char b[1024];
diff --git a/src/modules/module-protocol-pulse/pulse-server.c b/src/modules/module-protocol-pulse/pulse-server.c
index a416228..aafee18 100644
--- a/src/modules/module-protocol-pulse/pulse-server.c
+++ b/src/modules/module-protocol-pulse/pulse-server.c
@@ -63,6 +63,18 @@
 
 #define NAME	"pulse-server"
 
+/* strndupa may already be defined in another compatibility header */
+#if !defined(strndupa)
+#define strndupa(x_src, x_n) \
+        (__extension__ ( {   \
+                const char* x_in  = (x_src);                  \
+                size_t      x_len = strnlen(x_in, (x_n)) + 1; \
+                char*       x_out = (char *) alloca(x_len);   \
+                x_out[x_len-1] = '\0';                        \
+                (char *) memcpy(x_out, x_in, x_len-1);        \
+        } ) )
+#endif
+
 struct impl;
 struct server;
 
